README.md

```diff
@@ -1 +1,128 @@
+
+Writing modular code is the best practice because the more you can re-use the old code from other parts of the application in a new features and places, the better the overall project will be. There is one tiny problem with browser JavaScript however—it doesn't have built-in modules! That is insane if you think about it. Developers have to use HTML which is another language to include browser JavaScript files. But even then, the dependency management is awful and the `<script>` tags are often blocking the loading of other resources (they are synchronous).
+
+Ingenuity of the JavaScript community led to the invention of various JavaScript-only approaches to modules such as CommonJS, RequireJS, and AMD. Under the hood, these standards and libraries used AJAX (Asynchronous JavaScript and XML) or XHR to fetch other JavaScript libraries/modules. Not ideal, but works better than having to manage 100s of `<script>` tags.
+
+Of course, creators of Node.js new about this issue. They implemented CommonJS notation for the module system. It's very simple and elegant. 
+
+## Syntax for Working with Node Modules
+
+How do you get started with modules in Node? Simple. All you need to do is use `require` for importing from a main file, and `module.exports` for exporting in the module.
+
+Consider monolithic code (the code which has all the logic in it) which outputs hello in different languages:
+
+```js
+var sayHelloInEnglish = function() {
+  return 'Hello'
+}
+
+var sayHelloInTatar = function() {
+  return 'Isänmesez'
+}
+
+console.log('Tatar ' +
+  sayHelloInTatar() +
+  ' & English ' +
+  sayHelloInEnglish())
+```
+
+The code is in the `hello-monolithic.js` and if you run it with `$ node hello-monolithic.js`, you'll see this:
+
+```
+Tatar Isänmesez & English Hello
+```
+
+So far so good? What if you need to re-use the same hellos in multiple places? Let's abstract (separate the monolithic code into two files) into `greetings.js` (module) and `hello.js` (main file).
+
+The code for the `greetings.js` module  uses `module.exports`:
+
+```
+module.exports.sayHelloInEnglish = function() {
+  return 'Hello'
+}
+
+module.exports.sayHelloInTatar = function() {
+  return 'Isänmesez'
+}
+```
+
+Then in the main file `hello.js`, we import the `greetings.js` with require. Then we invoke the methods like any other functions:
+
+```js
+var greetings = require('./greetings.js')
+
+console.log('Tatar ' +
+  greetings.sayHelloInSwedish() +
+  ' & English ' +
+  greetings.sayHelloInEnglish())
+```
+
+
+## Exporting
+
+To export a module, we use `module.exports`. It's a global object meaning there's a property `module` on the `global` object. You can create a property on the `module.exports` object instead of overwriting it completely. 
```

Comment:
HERE show students how to export their code - and where to put it in greetings.js